tests/h/traversal/roots_test.py

```diff
@@ -497,17 +499,48 @@ def test_it_returns_users(self, factories, user_factory, user_service):
     def user_factory(self, pyramid_request):
         return UserRoot(pyramid_request)
 
-    @pytest.fixture
-    def user_service(self, pyramid_config):
-        user_service = mock.create_autospec(UserService, spec_set=True, instance=True)
-        pyramid_config.register_service(user_service, name="user")
-        return user_service
+
+@pytest.mark.usefixtures("user_service")
+class TestUserUserIDRoot(object):
+    def test_it_fetches_the_requested_user(
+        self, pyramid_request, user_factory, user_service
+    ):
+        user_factory["acct:bob@example.com"]
+
+        user_service.fetch.assert_called_once_with("acct:bob@example.com")
+
+    def test_it_raises_KeyError_if_the_user_does_not_exist(
+        self, user_factory, user_service
+    ):
+        user_service.fetch.return_value = None
+
+        with pytest.raises(KeyError):
+            user_factory["does_not_exist"]
+
+    def test_it_returns_UserContexts(self, factories, user_factory, user_service):
+        user_service.fetch.return_value = user = factories.User.build()
+
+        resource = user_factory[user.username]
+
+        assert isinstance(resource, UserContext)
 
     @pytest.fixture
-    def client_authority(self, patch):
-        client_authority = patch("h.traversal.roots.client_authority")
-        client_authority.return_value = None
-        return client_authority
+    def user_factory(self, pyramid_request):
```

Comment:
```suggestion
    def user_userid_root(self, pyramid_request):
```

We should keep the names of fixtures matching with the names of the real classes/variables/etc that they correspond to (which is our existing pattern in tests). Otherwise people make up all sorts of inconsistent fixture naming styles and it ends up making the code harder to read.